components/resourceDetails/index.jsx

```diff
@@ -92,6 +95,13 @@ export default function ResourceDetails({
     getAccordionKey(dataset, "sharing"),
     false
   );
+  const handleResourceCopyClick = () => {
+    navigator.clipboard.writeText(datasetUrl);
```

Comment:
You might wish to wrap this in a try/catch, though I'm sure this API is widely supported, I'm not sure on errors it could throw off top of head.